### PR TITLE
feat: add add-to-basket cli command

### PR DESCRIPTION
**Pull request by aleem1314**
## Description

Ref: #732 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/regen/ecocredit/basket/v1/types.proto [hunk 1]
```diff
@@ -15,3 +15,9 @@ message BasketCredit {
   //  credit type for this batch.
   string amount = 2;
 }
+
+// BasketCredits represents the slice of BasketCredit objects.
+message BasketCredits {
+  // credits is the list of basket credits
+  repeated BasketCredit credits = 1;
+}
```
Comment (ryanchristo) [inline proto/regen/ecocredit/basket/v1/types.proto:23]: I believe we want changes to proto definitions merged into master and backported to the release branch. We will need to add this addition to master as well, correct? @aaronc @technicallyty @robert-zaremba

**Review by ryanchristo — COMMENTED** [review]
Looks great. Changes to proto definitions are being applied to master as well. We might need to make those changes on master and backport or apply the same changes to master.

File: proto/regen/ecocredit/basket/v1/types.proto (hunk 1 above)
Comment (robert-zaremba) [reply proto/regen/ecocredit/basket/v1/types.proto:23]: I think we can do either way (unless there is a change in state.proto). Those backports needs to be done manually.

File: x/ecocredit/client/basket/tx.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+
+	keeper "github.com/regen-network/regen-ledger/x/ecocredit/basket"
+)
+
+func TxAddToBasket() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add-to-basket [owner] [basket_denom] [credits_json_file]",
```
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply x/ecocredit/client/basket/tx.go:17]: same for function name

File: x/ecocredit/client/basket/tx.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		Short: "add credits to the basket",
+		Long: strings.TrimSpace(`add credits to the basket.
+
+Parameters:
+		owner: account address of the owner of credits being put into the basket.
+		Note, the'--from' flag is ignored as it is implied from [owner]
+		basket_denom: basket denom is the basket denom to add credits to.
```
Comment (robert-zaremba) [inline x/ecocredit/client/basket/tx.go:24]:
```suggestion
		basket_denom: basket identifier
```

**Review by aaronc — APPROVED** [review]

**Pull request merged**